src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -59,23 +59,55 @@ public bool CanReplaceWithRValue(SemanticModel semanticModel, [NotNullWhen(true)
+        // Skip the tuple expression if the identifier is part of the simple identifier expression inside a tuple,
+        // but not the identifier of the field
+        // For example, we are evaluating `x` in `(x, y)`, or the expression after the colon in `(x: x, y: y)`
+        // but not the name of the field
+        // We want to handle the implicit declaration of the tuple field when finding its references, not via its name implication
+        if (sourceAncestor is IdentifierNameSyntax { Parent: ArgumentSyntax { Parent: TupleExpressionSyntax parentTuple } })
+        {
+            sourceAncestor = parentTuple.Parent;
+        }
+
+        if (sourceAncestor is null)
+            return null;
+
+        foreach (var ancestor in sourceAncestor.AncestorsAndSelf())
         {
             var symbol = semanticModel.GetDeclaredSymbol(ancestor, cancellationToken);
             if (symbol != null)
             {
-                if (symbol is IMethodSymbol { MethodKind: MethodKind.Conversion })
+                switch (symbol)
                 {
-                    // The token may be part of a larger name (for example, `int` in `public static operator int[](Goo g);`.
-                    // So check if the symbol's location encompasses the span of the token we're asking about.
-                    if (symbol.Locations.Any(static (loc, location) => loc.SourceTree == location.SourceTree && loc.SourceSpan.Contains(location.SourceSpan), location))
-                        return symbol;
-                }
-                else
-                {
-                    // For any other symbols, we only care if the name directly matches the span of the token
-                    if (symbol.Locations.Contains(location))
-                        return symbol;
+                    case IMethodSymbol { MethodKind: MethodKind.Conversion }:
+                        {
+                            // The token may be part of a larger name (for example, `int` in `public static operator int[](Goo g);`.
+                            // So check if the symbol's location encompasses the span of the token we're asking about.
+                            if (symbol.Locations.Any(static (loc, location) => loc.SourceTree == location.SourceTree && loc.SourceSpan.Contains(location.SourceSpan), location))
+                                return symbol;
+                            break;
+                        }
+
+                    case IPropertySymbol { ContainingType: ITypeSymbol { IsAnonymousType: true } }:
+                        {
+                            // If the span is part of the property's assignment expression, it's not our target result
+                            // For example, we are evaluating `a.Length` in `new { a.Length }`, or in `new { Length = a.Length }`
+                            // We want to handle the implicit declaration of the property when finding its references, not via its name implication
```

Comment:
I don't understand these comments about finding references. That's not what this function is for. It's not part of the find refs space. It's a general helper for lots of features